gateway/handle.go

```diff
@@ -684,72 +696,69 @@ func (gw *Handle) internalBatchHandlerFunc() http.HandlerFunc {
 		gw.stats.NewTaggedStat(
 			"gateway.write_key_failed_requests",
 			stats.CountType,
-			gw.newSourceStatTagsWithReason(arctx, reqType, errorMessage),
+			gw.newReqTypeStatsTagsWithReason(reqType, errorMessage),
 		).Increment()
 		gw.logger.Infon("response",
 			logger.NewStringField("ip", kithttputil.GetRequestIP(r)),
 			logger.NewStringField("path", r.URL.Path),
 			logger.NewIntField("status", int64(status)),
 			logger.NewStringField("body", responseBody),
 		)
+		gw.logger.Debugn("response",
+			logger.NewStringField("ip", kithttputil.GetRequestIP(r)),
+			logger.NewStringField("path", r.URL.Path),
+			logger.NewIntField("status", int64(status)),
+			logger.NewStringField("body", responseBody),
+			logger.NewStringField("request", string(body)),
+		)
 		http.Error(w, responseBody, status)
 	}
 }
 
-func (gw *Handle) extractJobsFromInternalBatchPayload(
-	arctx *gwtypes.AuthRequestContext,
-	reqType string,
-	body []byte,
-) ([]*jobsdb.JobT, error) {
-	if !gjson.ValidBytes(body) {
-		return nil, fmt.Errorf("%s", response.InvalidJSON)
+func (gw *Handle) extractJobsFromInternalBatchPayload(reqType string, body []byte) ([]*jobsdb.JobT, error) {
+	type params struct {
+		MessageID       string `json:"message_id"`
+		SourceID        string `json:"source_id"`
+		SourceJobRunID  string `json:"source_job_run_id"`
+		SourceTaskRunID string `json:"source_task_run_id"`
+		UserID          string `json:"user_id"`
+		TraceParent     string `json:"traceparent"`
 	}
-	gw.requestSizeStat.Observe(float64(len(body)))
 
-	type jobObject struct {
-		userID     string
-		events     []map[string]interface{}
-		receivedAt string
+	type singularEventBatch struct {
+		Batch      []json.RawMessage `json:"batch"`
+		ReceivedAt string            `json:"receivedAt"`
 	}
```

Comment:
We can take ip as well. It's been added now in rudder schema.
```suggestion
		type SingularEventBatch struct {
			Batch      []map[string]interface{} `json:"batch"`
			RequestIP  string                   `json:"requestIP"`
			ReceivedAt string                   `json:"receivedAt"`
		}
```